### ISSUE TITLE
Extension can not start

### ISSUE THREAD

**Issue by silv7k** [ThosRTanner/inforss#186]
I’ll quote my post from Palemoon forum:

From time to time when I restart a browser (this may happen once in a week. or a month) infoRSS Reloaded couldn’t start.

As far as I can tell the extension can not read (or parse) RSS list, I get

> rssFillPopup : TypeError: RSSList is null

popup alert trying to click the extension button in the status bar.

If I rightclick the button to go to configuration I get

> redisplay_configuration : TypeError: gInforssMediator is null

popup alert, then Options window is open with no information at all (no feeds, no groups, no [old] settings…).

Mostly I fix it like this: enable the old infoRSS extension, restart browser, enable infoRSS Reloaded, restart browser. After this “fix” everything seems to work fine for a time.

This is an old story, I believe it has nothing to do with PM 28 and infoRSS Reloaded 2.0.3.2 specifically (symptoms are mostly the same every time).

Here goes error log (filtered by ’inforss’):
`01:06:28.108 [InfoRSS] failed to start:  TypeError: addon is null
Stack trace:
get_name@chrome://inforss/content/modules/Version.jsm:126:5
alert@chrome://inforss/content/modules/Prompt.jsm:61:31
checkContentHandler@chrome://inforss/content/inforss.js:263:5
inforssStartExtension@chrome://inforss/content/inforss.js:106:7
  
inforss.js:146
01:16:21.184 Exception in redisplay_configuration TypeError: gInforssMediator is null
Stack trace:
redisplay_configuration@chrome://inforss/content/inforssOption.js:159:5
load_and_display_configuration@chrome://inforss/content/inforssOption.js:147:3
init@chrome://inforss/content/inforssOption.js:135:5
onload@chrome://inforss/content/inforssOption.xul:1:1
cmd_showItemPreferences_doCommand@chrome://mozapps/content/extensions/extensions.js:1066:9
gVC_doCommand@chrome://mozapps/content/extensions/extensions.js:1350:5
showPreferences@chrome://mozapps/content/extensions/extensions.xml:1631:11
oncommand@about:addons:1:1
  
Debug.jsm:155
01:18:38.758 Exception in rssFillPopup TypeError: RSSList is null
Stack trace:
get@chrome://inforss/content/inforssXMLRepository.js:396:9
rssFillPopup@chrome://inforss/content/inforss.js:528:1
onpopupshowing@chrome://browser/content/browser.xul:1:39
  
Debug.jsm:155

I also attach my inforss.xml

**Comment by ThosRTanner** [comment]
This:

```
01:06:28.108 [InfoRSS] failed to start:  TypeError: addon is null
Stack trace:
get_name@chrome://inforss/content/modules/Version.jsm:126:5
alert@chrome://inforss/content/modules/Prompt.jsm:61:31
checkContentHandler@chrome://inforss/content/inforss.js:263:5
inforssStartExtension@chrome://inforss/content/inforss.js:106:7
```

Is very strange.

**Comment by JustOff** [comment]
That's quite possible if `get_version()` called too early, since `AddonManager.getAddonByID()` is async.

**Comment by silv7k** [comment]
The most recent error log entry I have (after all these restarts) is exactly the same.

**Comment by ThosRTanner** [comment]
Ugghh. It's really async? How is an addon meant to find out about itself in any sane way?

**Comment by JustOff** [comment]
By running the init from callback of `AddonManager.getAddonByID()`

**Comment by ThosRTanner** [comment]
yeah its just i now end up with a callback (window.onload) which needs to provide a callback to a callback. ugh.

**Comment by JustOff** [comment]
Welcome to the world of async ;-) AFAIK, any other ways leads to the race conditions sooner or later ...

**Comment by ThosRTanner** [comment]
it's sad that there isn't a way of an extension to find out information about itself though. if I ever change the ID of this extension, I'll have to change that bit of code.

**Comment by silv7k** [comment]
This version starts OK (4 times out of 4 restarts).
It starts and functions as intended.
